test/e2e/gameserver_test.go

```diff
@@ -211,6 +211,27 @@ func TestGameServerUnhealthyAfterDeletingPod(t *testing.T) {
 	assert.NoError(t, err)
 }
 
+func TestGameServerUnhealthyAfterReadyCrash(t *testing.T) {
+	t.Parallel()
+	gs := defaultGameServer()
+	readyGs, err := framework.CreateGameServerAndWaitUntilReady(defaultNs, gs)
+	if err != nil {
+		t.Fatalf("Could not get a GameServer ready: %v", err)
+	}
+
+	gsClient := framework.AgonesClient.AgonesV1().GameServers(defaultNs)
+	defer gsClient.Delete(readyGs.ObjectMeta.Name, nil) // nolint: errcheck
+
+	// there is no response, so don't wait for the timeout
+	go func() {
+		// ignore the error, we don't care about the response
```

Comment:
Is there any chance that this call fails? If so this test will become flaky. If you look at the response and retry on errors (until the server acks the crash) then the test will be more reliable.

Reply:
Admittedly it's a bit hacky. What I'm doing is ignoring the error that will come back when we don't get a response, because I didn't feel like writing a separate method to just send, and not wait for an ACK response.

While it's super unlikely there will fail (we use this all through e2e with no retries), I will admit to cutting a couple of corners here :confused: 

I'll update the test. I can't look at myself in the mirror anymore out of embarrassment :smile:

Reply:
Of course, now I've updated it, it won't pass... how very strange.